app/javascript/onboarding/components/IntroSlide.jsx

```diff
@@ -2,27 +2,98 @@ import { h, Component } from 'preact';
 import PropTypes from 'prop-types';
 
 import Navigation from './Navigation';
-import { userData, updateOnboarding } from '../utilities';
+import { getContentOfToken, userData, updateOnboarding } from '../utilities';
 
+/* eslint-disable camelcase */
 class IntroSlide extends Component {
   constructor(props) {
     super(props);
 
+    this.handleChange = this.handleChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
     this.user = userData();
+
+    this.state = {
+      checked_code_of_conduct: false,
+      checked_terms_and_conditions: false,
+      text: null,
+    };
   }
 
   componentDidMount() {
-    updateOnboarding('intro slide');
+    updateOnboarding('v2: intro, code of conduct, terms & conditions');
   }
 
   onSubmit() {
     const { next } = this.props;
-    next();
+    const csrfToken = getContentOfToken('csrf-token');
+
+    fetch('/onboarding_checkbox_update', {
+      method: 'PATCH',
+      headers: {
+        'X-CSRF-Token': csrfToken,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user: this.state }),
+      credentials: 'same-origin',
+    }).then((response) => {
+      if (response.ok) {
+        localStorage.setItem('shouldRedirectToOnboarding', false);
+        next();
+      }
+    });
+  }
+
+  handleChange(event) {
+    const { name } = event.target;
+    this.setState((currentState) => ({
+      [name]: !currentState[name],
+    }));
+  }
+
+  handleShowText(event, id) {
+    event.preventDefault();
+    this.setState({ text: document.getElementById(id).innerHTML });
+  }
+
+  buttonIsDisabled() {
+    const {
+      checked_code_of_conduct,
+      checked_terms_and_conditions,
+    } = this.state;
+
+    return !checked_code_of_conduct || !checked_terms_and_conditions;
+  }
+
+  backToSlide() {
```

Comment:
This appears to be used in only one place. It may not merit being a function. Also, what does setting the `text` state property to `false` do exactly when things rerender? It looks likes it closes the terms and conditions?

Reply:
This was preexisting in the original component I transferred it from...but I can render the function's contents inline since you're right that it's only used in one place.

I actually think, now that I'm looking at this a second time around, that the `text` property should be set to `null` rather than `false`. When it is set to `false` on the state, the component re-`render`s, and the [coc/terms + conditions view](https://github.com/thepracticaldev/dev.to/pull/7294/files#diff-414ac17a58f5d19e6d0d81e26c3ff5d5R81-R96) is no longer shown.